lib/services/services.c

```diff
@@ -1265,24 +1265,33 @@ resources_agent_exists(const char *standard, const char *provider, const char *a
  * \param[out] action        Where to set action result
  * \param[in]  agent_status  Exit status to set
  * \param[in]  exec_status   Execution status to set
- * \param[in]  reason        Human-friendly description of event to set
+ * \param[in]  format        printf-style format for a human-friendly
+ *                           description of reason for result
+ * \param[in]  ...           arguments for \p format
  */
 void
 services__set_result(svc_action_t *action, int agent_status,
-                     enum pcmk_exec_status exec_status, const char *reason)
+                     enum pcmk_exec_status exec_status, const char *format, ...)
 {
+    va_list ap;
+    int len = 0;
+    char *reason = NULL;
+
     if (action == NULL) {
         return;
     }
 
     action->rc = agent_status;
     action->status = exec_status;
 
-    if (!pcmk__str_eq(action->opaque->exit_reason, reason,
-                      pcmk__str_none)) {
-        free(action->opaque->exit_reason);
-        action->opaque->exit_reason = (reason == NULL)? NULL : strdup(reason);
+    if (format != NULL) {
+        va_start(ap, format);
+        len = vasprintf(&reason, format, ap);
```

Comment:
we could potentially shortcut in case format == '%s' and old-exit-reason !=NULL by chopping off the first argument
and comparing it to the previous string.
probably no way to avoid ugliness of having 2 different vasprintf calls then and overall questionable if worth the effort.

Reply:
creating a separate function would avoid the need